### ISSUE TITLE
Support statically-resolved configurations in the configurations block

### ISSUE THREAD

**Issue by ianbrandt** [gradle/kotlin-dsl#1118]
Despite applying the Kotlin plugin declaratively in the `plugins {}` block:

```
plugins {
	kotlin("jvm") version "1.2.70"
}
```

...I can only resolve configurations in the `configurations {}` block dynamically:

```
configurations {
	getByName("implementation") {
		exclude(group = "org.foo")
	}
}
```

The [migration guide](https://guides.gradle.org/migrating-build-logic-from-groovy-to-kotlin/#configurations-and-dependencies) says:

> Each configuration contributed by an applied plugin is also available as a member of the configurations container, so you can reference it just like any other configuration.
>
> *Knowing what configurations are available*
> The easiest way to find out what configurations are available is by asking your IDE for suggestions within the configurations container.
>
> You can also use the `:kotlinDslAccessorsReport` task, which prints the Kotlin code for accessing the configurations contributed by applied plugins and provides the names for all of those accessors.

Indeed the `:kotlinDslAccessorsReport` task prints:

```
/**
 * The 'implementation' configuration.
 */
val ConfigurationContainer.`implementation`: Configuration
    get() = getByName("implementation")
```

I see with Quick Documentation in IntelliJ that `dependencies` has a Kotlin DSL `Project` extension function:

> org.gradle.kotlin.dsl ProjectExtensionsKt.class
> public fun Project.dependencies(
>    configuration: DependencyHandlerScope.() → Unit
>): Unit

...whereas `configurations` does not:

> org.gradle.api.Project
> public abstract org.gradle.api.artifacts.ConfigurationContainer getConfigurations()

It would be great if the `implementation`, etc. extension properties were statically available in `configurations {}` as they are in `dependencies {}`):

```
configurations {
	implementation {
		exclude(group = "org.foo")
	}
}
```

**Comment by ianbrandt** [comment]
Slack thread: https://kotlinlang.slack.com/archives/C19FD9681/p1537046623000100

**Comment by eskatos** [comment]
Closing as fixed in #1129
